crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -887,7 +906,12 @@ where
                 InitializedArchiver {
                     archiver,
                     best_archived_block: (best_archived_block_hash, best_archived_block_number),
-                } = initialize_archiver(&segment_headers_store, &subspace_link, client.as_ref())?;
+                } = initialize_archiver(
+                    &segment_headers_store,
+                    &subspace_link,
+                    client.as_ref(),
+                    force_mapping,
```

Comment:
I wonder if this should always be `false`, so that we pick up the mappings in any partial block that hasn't already been archived?

(If there is a further gap, and more blocks end up being skipped, then we'll re-initialise the archiver without creating or sending those mappings.)